pkg/reconciler/internal/hook/hook.go

```diff
@@ -37,18 +37,22 @@ import (
 	"github.com/joelanford/helm-operator/pkg/manifestutil"
 )
 
-func NewDependentResourceWatcher(c controller.Controller, rm meta.RESTMapper) hook.PostHook {
+func NewDependentResourceWatcher(c controller.Controller, rm meta.RESTMapper, watchReleaseResources bool, extraWatches ...schema.GroupVersionKind) hook.PostHook {
```

Comment:
I think it is confusing to have `watchReleaseResources` and `extraWatches` as parameters in one function and letting the caller decide which one to choose. 

Imho adding two functions with different names delegating to a private `newDependentResourceWatcher` is preferred by me.

```
// NewDependentResourceWatcherForResources adds depended watches on specific configured resources. 
func NewDependentResourceWatcherForResources(c controller.Controller, rm meta.RESTMapper, watches ...schema.GroupVersionKind) hook.PostHook {
  return newDependentResourceWatcher(c, rm, false, watches)
}

// NewDependentReleaseResourceWatcher adds watchers to all objects created by the Helm release.
func NewDependentReleaseResourceWatcher(c controller.Controller, rm meta.RESTMapper) hook.PostHook {
  newDependentResourceWatcher(c, rm, true)
}
```

Writing documentation for it is imho more complex and involves mental overhead to understand the concept to why I should add `extraWatches` if I already added all  watched dependencies by setting `watchReleaseResources = true`.

Reply:
But with this change we can no longer separate the "skip dependent watches" and "watch extra resources" parameters. This has to be a single watcher, we cannot just call this function once with `watchReleaseResources = true` and then with the extra watches; if we ever go back to not skipping dependent watches, this would cause two reconciliations for each update to the resources specified here. Maybe not a concern because we would just drop the extra watches then, but it does complicate the option design substantially.

Reply:
Agree!